neqo-bin/src/client/http09.rs

```diff
@@ -76,7 +76,8 @@ impl<'a> super::Handler for Handler<'a> {
                 }
                 ConnectionEvent::StateChange(
                     State::WaitInitial | State::Handshaking | State::Connected,
-                ) => {
+                )
+                | ConnectionEvent::ZeroRttRejected => {
                     qdebug!("{event:?}");
                     self.download_urls(client);
```

Comment:
I don't think that this is correct.  You will need to reset the state of each of the streams as well.

My understanding is that if we have sent a request, we track that.  So if a request is partially sent, we need to resend.

(Maybe h09 doesn't do that, but h3 might have to.)